### PR TITLE
Prefill logprobs

### PR DESCRIPTION
**Pull request by xiaoguoguo626807**
### PR Category

Execute Infrastructure 

### PR Types

Improvements

### Description

支持返回prefill 阶段的logprobs

### 是否引起精度变化

 否

### CODE REVIEW

File: src/paddlefleet/generation/greedy_generator.py [hunk 1]
```diff
@@ -716,11 +787,19 @@ def generate(
...
+                    logprob_start_len,
+                    prompt_len,
+                    log_probs_per_batch,
+                )
             last_logits = logits[:, -1]  # [B, vocab]
             next_tok = _sample_next_token(
                 last_logits, temperature, top_k, top_p
             )
-            if return_log_probs and log_probs_per_batch is not None:
+            if return_log_probs and log_probs_per_batch is not None and True:
```
Comment (risemeup1111) [inline src/paddlefleet/generation/greedy_generator.py]: P1 这里无条件把 prefill 产生的第一个新 token 追加到结果，下面的 decode 分支也对后续 token 无条件追加。因此当 `logprob_start_len > prompt_len` 时，返回值仍从 `prompt_len` 开始，违背接口“从绝对位置开始”的文档，新增的 `test_start_beyond_prompt_skips_generated_tokens` 在缓存和无缓存路径都会失败。请按当前 token 的绝对位置（prefill 为 `prompt_len`，decode 逐步递增）仅在达到 `logprob_start_len` 后追加，并统一修正两条路径。

File: src/paddlefleet/generation/greedy_generator.py [hunk 2]
```diff
@@ -627,8 +693,11 @@ def generate(
 
         bsz, prompt_len = input_ids.shape
         generated = input_ids.clone()
+        logprob_start_len = _resolve_logprob_start_len(
```
Comment (risemeup1111) [inline src/paddlefleet/generation/greedy_generator.py]: P2 文档声明 `logprob_start_len` 在 `return_log_probs=False` 时会被忽略，但这里无条件调用 resolver；因此即使不请求 log-probs，传入 `logprob_start_len=-1` 仍会抛出 `ValueError`，改变了该调用的行为。请仅在启用 `return_log_probs` 时解析/校验该参数，或在关闭时直接跳过。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
| 序号 | 位置 | 优先级 | 规则来源 | 状态 |
| --- | --- | --- | --- | --- |
| 1 | [生成 token 的 log-prob 窗口](https://github.com/PaddlePaddle/PaddleFleet/pull/1816#discussion_r3827016732) | P1 | [仓库规则：功能正确性与兼容性](https://github.com/PaddlePaddle/PaddleFleet/blob/1810e66544c146525256c74db507f985d908d35f/.agents/skills/ai-review/references/base-rules.md) | ✅ |
| 2 | [关闭日志时的参数校验](https://github.com/PaddlePaddle/PaddleFleet/pull/1816#discussion_r3827016816) | P2 | [仓库规则：功能正确性与兼容性](https://github.com/PaddlePaddle/PaddleFleet/blob/1810e66544c146525256c74db507f985d908d35f/.agents/skills/ai-review/references/base-rules.md) | ✅ |

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.6-sol 默认推理级别, please check the suggestions carefully.</sup>
</div>

File: src/paddlefleet/generation/greedy_generator.py (hunk 1 above)
Comment (risemeup1111) [reply src/paddlefleet/generation/greedy_generator.py]: 作者最新提交仅移除了冗余条件，生成 token 的追加逻辑仍未按 `logprob_start_len` 过滤；当前 head 该问题仍未解决，保持 P1。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]

File: src/paddlefleet/generation/greedy_generator.py (hunk 1 above)
Comment (risemeup1111) [reply src/paddlefleet/generation/greedy_generator.py]: 当前 head 已在缓存、无缓存及 decode 路径按生成 token 的绝对位置过滤 log-prob；`logprob_start_len > prompt_len` 时会跳过前置生成 token，该问题已修复。

**Review by risemeup1111 — COMMENTED** [review]
当前 head 已修复 P1 的生成 token log-prob 窗口问题；P2 的参数校验问题仍然存在：`return_log_probs=False` 时仍会解析 `logprob_start_len`，请按既有线程处理。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.6-sol 默认推理级别, please check the suggestions carefully.</sup>
</div>

File: src/paddlefleet/generation/greedy_generator.py (hunk 2 above)
Comment (risemeup1111) [reply src/paddlefleet/generation/greedy_generator.py]: 当前 head 已将 `_resolve_logprob_start_len` 放入 `return_log_probs=True` 分支；关闭 log-probs 时该参数不再校验，P2 问题已修复。

**Review by risemeup1111 — APPROVED** [review]

**Comment by liuhao2638** [comment]
@risemeup1111 分析CI报错原因

**Comment by risemeup1111** [comment]
@liuhao2638 已核对本次 CI 日志，失败原因与本 PR 的 `greedy_generator.py` 改动没有直接关联：

- 单卡单测达到 GitHub Actions 120 分钟超时，日志末尾停在 TileLang kernel 编译，未见失败断言。
- 多卡单测仅明确失败于 `tests/multi_card_tests/transformer/test_kimi_delta_attention_cp.py`：rank 3 在 `causal_conv1d_cp` 的 `dist.all_gather` 报 NCCL `unhandled cuda error`。
- H20 多卡集成的失败子任务出现 pod 被终止、`Exit code -15` 及 NCCL `unhandled cuda error`，属于运行环境/分布式资源侧失败信号。
- H20 model accuracy alignment 在 `AutoConfig.from_pretrained` 处理本地缓存路径 `/home/.cache/PaddleFormers/MiniMax-V2.5-bf16_2EP` 时触发 `huggingface_hub.errors.HFValidationError`，路径被当作 Hugging Face repo id 校验。

H20 单卡和 A100 集成已通过。建议先重试超时/NCCL 相关任务，并修正 alignment job 的本地模型路径配置。

**Comment by liuhao2638** [comment]
@risemeup1111  除了精度 Not equal to GT的，单测有哪些挂的。除了精度equal还有没有别的问题

**Comment by risemeup1111** [comment]
已核对 workflow `32698701214`。除 accuracy alignment 的 `Not equal to GT` 外，已确认的失败还有：

- 单卡单测：上一轮达到 GitHub Actions 120 分钟超时，日志停在 TileLang kernel 编译，未见失败断言；本轮 job 仍在重跑。
- 多卡单测：`tests/multi_card_tests/transformer/test_kimi_delta_attention_cp.py` 的 rank 3 在 `causal_conv1d_cp` 的 `dist.all_gather` 报 NCCL `unhandled cuda error`；本轮 job 仍排队。
- H20 多卡集成：`GLM4.5 pre-train (EP4)` 的 rank 3 以 `Exit code -15` 退出并伴随 NCCL `unhandled cuda error`；`Qwen lora` 的 rank 0 触发 libuv `uv__finish_close` 断言，因未产生 train metrics 判定失败。
- 上一轮 accuracy alignment 还出现过 `AutoConfig.from_pretrained` 将本地缓存路径当作 Hugging Face repo id 的 `HFValidationError`；本轮 alignment 仍排队，尚不能判断是否已恢复。

因此，除了精度对齐外确有上述单测/集成失败信号，但目前都更像超时、分布式资源或作业清理问题；不能据此归因到本 PR 的 `greedy_generator.py` 改动。H20 单卡和 A100 集成已通过。

**Review by changeyoung98 — APPROVED** [review]
